tests/ui/editions/dyn-trait-sugg-2021.stderr

```diff
@@ -2,7 +2,7 @@ error[E0782]: trait objects must include the `dyn` keyword
   --> $DIR/dyn-trait-sugg-2021.rs:10:5
    |
 LL |     Foo::hi(123);
-   |     ^^^
+   |     ^^^ no desired associated item found in this trait, if you do not want to use a bare trait object here
```

Comment:
If this is the only suggestion I don't think it helps much. It's probably better to first do something different for when `<dyn A as B>::something` and `dyn A: B` does not hold. Because in that case we are certain that the user didn't want `A::something` to mean `<dyn A as B>::something`.